ir/src/test/scala/org/scalajs/core/ir/PrintersTest.scala

```diff
@@ -0,0 +1,999 @@
+    assertPrintEquals("(x ^[long] y)",
+        BinaryOp(Long_^, ref("x", LongType), ref("y", LongType)))
+    assertPrintEquals("(x <<[long] y)",
+        BinaryOp(Long_<<, ref("x", LongType), ref("y", IntType)))
+    assertPrintEquals("(x >>>[long] y)",
+        BinaryOp(Long_>>>, ref("x", LongType), ref("y", IntType)))
+    assertPrintEquals("(x >>[long] y)",
+        BinaryOp(Long_>>, ref("x", LongType), ref("y", IntType)))
+
+    assertPrintEquals("(x ==[bool] y)",
+        BinaryOp(Boolean_==, ref("x", BooleanType), ref("y", BooleanType)))
+    assertPrintEquals("(x !=[bool] y)",
+        BinaryOp(Boolean_!=, ref("x", BooleanType), ref("y", BooleanType)))
+    assertPrintEquals("(x |[bool] y)",
+        BinaryOp(Boolean_|, ref("x", BooleanType), ref("y", BooleanType)))
+    assertPrintEquals("(x &[bool] y)",
+        BinaryOp(Boolean_&, ref("x", BooleanType), ref("y", BooleanType)))
+  }
+
+  @Test def printNewArray(): Unit = {
+    assertPrintEquals("new I[3]", NewArray(ArrayType("I", 1), List(i(3))))
+    assertPrintEquals("new I[3][]", NewArray(ArrayType("I", 2), List(i(3))))
+    assertPrintEquals("new O[3][4][][]",
+        NewArray(ArrayType("O", 4), List(i(3), i(4))))
+  }
+
+  @Test def printArrayValue(): Unit = {
+    assertPrintEquals("I[]()",
+        ArrayValue(ArrayType("I", 1), List()))
+    assertPrintEquals("I[](5, 6)",
+        ArrayValue(ArrayType("I", 1), List(i(5), i(6))))
+
+    assertPrintEquals("I[][](null)",
+        ArrayValue(ArrayType("I", 2), List(Null())))
+  }
+
+  @Test def printArrayLength(): Unit = {
+    assertPrintEquals("x.length", ArrayLength(ref("x", ArrayType("I", 1))))
+  }
+
+  @Test def printArraySelect(): Unit = {
+    assertPrintEquals("x[3]",
+        ArraySelect(ref("x", ArrayType("I", 1)), i(3))(IntType))
+  }
+
+  @Test def printRecordValue(): Unit = {
+    assertPrintEquals("(x = 3, y = 4)",
+        RecordValue(
+            RecordType(List(
+                RecordType.Field("x", None, IntType, mutable = false),
+                RecordType.Field("y", None, IntType, mutable = true))),
+            List(i(3), i(4))))
+  }
+
+  @Test def printIsInstanceOf(): Unit = {
+    assertPrintEquals("x.isInstanceOf[T]",
+        IsInstanceOf(ref("x", AnyType), StringClass))
+  }
+
+  @Test def printAsInstanceOf(): Unit = {
+    assertPrintEquals("x.asInstanceOf[T]",
+        AsInstanceOf(ref("x", AnyType), StringClass))
+  }
+
+  @Test def printUnbox(): Unit = {
+    assertPrintEquals("x.asInstanceOf[I]", Unbox(ref("x", AnyType), 'I'))
+  }
+
+  @Test def printGetClass(): Unit = {
+    assertPrintEquals("x.getClass()", GetClass(ref("x", AnyType)))
+  }
+
+  @Test def printCallHelper(): Unit = {
+    assertPrintEquals("help(x, y)",
+        CallHelper("help", List(ref("x", AnyType), ref("y", AnyType)))(IntType))
+  }
+
+  @Test def printJSNew(): Unit = {
+    assertPrintEquals("new C()", JSNew(ref("C", AnyType), Nil))
+    assertPrintEquals("new C(4, 5)", JSNew(ref("C", AnyType), List(i(4), i(5))))
```

Comment:
Could you also add one that exercises the nesting of `containsOnlySelectsFromAtom`?